packages/js/product-editor/src/data-form/components/fields/name/field.tsx

```diff
@@ -0,0 +1,214 @@
+		error: nameValidationError,
+		validate: validateName,
+	} = useValidation< Product >(
+		'name',
+		async function nameValidator() {
+			if ( ! name || name === AUTO_DRAFT_NAME ) {
+				return {
+					message: __( 'Product name is required.', 'woocommerce' ),
+				};
+			}
+
+			if ( name.length > 120 ) {
+				return {
+					message: __(
+						'Please enter a product name shorter than 120 characters.',
+						'woocommerce'
+					),
+				};
+			}
+		},
+		[ name ]
+	);
+
+	const setSkuIfEmpty = () => {
+		if ( sku || nameValidationError ) {
+			return;
+		}
+		onChange( { sku: cleanForSlug( name ) } );
+	};
+
+	const help =
+		nameValidationError ??
+		( productId &&
+			[ 'publish', 'draft' ].includes( data.status ) &&
+			permalinkPrefix && (
+				<span className="woocommerce-product-form__secondary-text product-details-section__product-link">
+					{ __( 'Product link', 'woocommerce' ) }
+					:&nbsp;
+					<a href={ data.permalink } target="_blank" rel="noreferrer">
+						{ permalinkPrefix }
+						{ data.slug || cleanForSlug( name ) }
+						{ permalinkSuffix }
+					</a>
+					<Button
+						variant="link"
+						onClick={ () => setShowProductLinkEditModal( true ) }
+					>
+						{ __( 'Edit', 'woocommerce' ) }
+					</Button>
+				</span>
+			) );
+
+	const nameControlId = useInstanceId(
+		BaseControl,
+		'product_name'
+	) as string;
+
+	// @todo: This relies on attribute settings from the block information, we need to find a way to get this into the DataForm.
+	// useEffect( () => {
+	// 	if ( field.attributes.autoFocus ) {
+	// 		selectBlock( clientId );
+	// 	}
+	// }, [] );
+
+	function handleSuffixClick() {
+		onChange( { featured: ! featured } );
+	}
+
+	function renderFeaturedSuffix() {
+		const markedText = __( 'Mark as featured', 'woocommerce' );
+		const unmarkedText = __( 'Unmark as featured', 'woocommerce' );
+		const tooltipText = featured ? unmarkedText : markedText;
+
+		return (
+			<Tooltip text={ tooltipText } placement="top">
+				{ featured ? (
+					<Button
+						icon={ starFilled }
+						aria-label={ unmarkedText }
+						onClick={ handleSuffixClick }
+					/>
+				) : (
+					<Button
+						icon={ starEmpty }
+						aria-label={ markedText }
+						onClick={ handleSuffixClick }
+					/>
+				) }
+			</Tooltip>
+		);
+	}
+
+	return (
+		<div>
+			<BaseControl
+				id={ nameControlId }
+				label={
+					<Label label={ __( 'Name', 'woocommerce' ) } required />
+				}
+				className={ classNames( {
+					'has-error': nameValidationError,
+				} ) }
+				help={ help }
+			>
+				<InputControl
+					id={ nameControlId }
+					ref={ nameRef }
+					name="name"
+					// eslint-disable-next-line jsx-a11y/no-autofocus
+					// autoFocus={ attributes.autoFocus }
+					placeholder={ __( 'e.g. 12 oz Coffee Mug', 'woocommerce' ) }
+					onChange={ ( nextValue ) => {
+						onChange( { name: nextValue ?? '' } );
+					} }
+					value={ name && name !== AUTO_DRAFT_NAME ? name : '' }
+					autoComplete="off"
```

Comment:
qq: Is there any reason for setting auto complete to off?

Reply:
I found the PR which introduced it: https://github.com/woocommerce/woocommerce/pull/39211

TBH, I am not convinced if we should disable the autocomplete. But yeah, I think we can discuss it later.